core/includes/menu.inc

```diff
@@ -3495,13 +3495,19 @@ function _menu_clear_page_cache() {
   // Clear the page and block caches, but at most twice, including at
   //  the end of the page load when there are multiple links saved or deleted.
   if ($cache_cleared == 0) {
-    cache_clear_all();
+    if (module_exists('layout')) {
```

Comment:
Layout is a required module (`required = TRUE` in info file), so why checking that it's on, anyway? It can't be off.

That would halve the size of this PR... :grin: